Lib/fontbakery/profiles/name.py

```diff
@@ -261,39 +261,75 @@ def com_google_fonts_check_monospace(ttFont, glyph_metrics_stats):
-        # with only the first occurences of the name entries
-        # will suffice:
-        fullfontname = fullfontname[0]
-        familyname = familyname[0]
-
-        if not fullfontname.startswith(familyname):
-            yield FAIL,\
-                  Message("does-not",
-                          f"On the 'name' table, the full font name"
-                          f" (NameID {NameID.FULL_FONT_NAME}"
-                          f" - FULL_FONT_NAME: '{familyname}')"
-                          f" does not begin with font family name"
-                          f" (NameID {NameID.FONT_FAMILY_NAME}"
-                          f" - FONT_FAMILY_NAME: '{fullfontname}')")
-        else:
-            yield PASS, "Full font name begins with the font family name."
+    name_table = ttFont["name"]
+    names_compared = False  # Used for knowing if at least one comparison was attempted
+    passed = True
+
+    # Collect all the unique platformIDs, encodingIDs, and languageIDs
+    # used in the font's name table.
+    platform_ids = set(rec.platformID for rec in name_table.names)
+    encoding_ids = set(rec.platEncID for rec in name_table.names)
+    language_ids = set(rec.langID for rec in name_table.names)
+
+    # Now iterate over the platform/encoding/languageID sets and compare
+    # the full name and family name strings.
+    # The full name string is always taken from nameID 4, but
+    # the family name string can come from nameIDs 1, 16, or 21.
+    # We'll compare all of the ones contained in the font.
+    full_name_id = NameID.FULL_FONT_NAME
+    for plat_id in sorted(platform_ids):
+        for enc_id in sorted(encoding_ids):
+            for lang_id in sorted(language_ids):
+
+                # Check if the full name record exists
+                if name_table.getName(full_name_id, plat_id, enc_id, lang_id) is None:
+                    # The full_name_id wasn't found. No point in going further
+                    continue
+
+                # Iterate over all possible family name records
+                for family_name_id in (NameID.FONT_FAMILY_NAME,
+                                       NameID.TYPOGRAPHIC_FAMILY_NAME,
+                                       NameID.WWS_FAMILY_NAME):
+                    if name_table.getName(
+                            family_name_id, plat_id, enc_id, lang_id) is None:
+                        # The family_name_id wasn't found. Move on to the next
+                        continue
+
+                    names_compared = True  # Yay! At least one comparison will be made!!
+
+                    try:
+                        family_name = name_table.getName(
+                            family_name_id, plat_id, enc_id, lang_id).toUnicode()
+                        full_name = name_table.getName(
+                            full_name_id, plat_id, enc_id, lang_id).toUnicode()
+                    except UnicodeDecodeError as err:
+                        raise err
```

Comment:
why would this try/except be needed here?

Reply:
For the same reason it's used [here](https://github.com/fonttools/fonttools/blob/41bd3a143d3182ccb5604b7b01d1cf088da75e94/Lib/fontTools/ttLib/tables/_n_a_m_e.py#L107-L110): decoding of the name record's string could fail.

Reply:
In the fonttools code you referenced, the except block has a `continue` keyword, which makes sense.

But here, you have `raise err` which (if I'm not mistaken in my analysis) will cause the exception to propagate and result in the check emitting an **ERROR** status.

Reply:
You analysis is correct.

By writing that FontTools solution _makes sense_ are you saying that you'd rather skip a name record whose string cannot be decoded, instead of letting the user know —by means of the ERROR that will be yielded— that there's something wrong with their font's name table?
You should take into account that that FontTools' snippet is part of a method that includes a fallback mechanism which will always return something, including `None`. Such pattern doesn't really work for this check.

As a Font Bakery **user**, my preference goes to the currently implemented solution because I want to know that something is wrong with my font, even if the check didn't succeed because of things that it's not meant to check for.

But I'm not against changing the `raise` to a `continue` if you think that handling is more in line with FB's mantra.
Let me know.

Reply:
You know what? Going to make the change to `continue`, and write a new check whose goal is to validate that all name table strings can be decoded. What do you think?

Reply:
Font Bakery's policy is that ERROR means either (a) bad setup in the user's environment such as a missing dependency or (b) a bug in the code that must be fixed.

For problems in a font we always use a FAIL or a WARN instead.

So I believe that a good solution here would be to yield a FAIL or WARN here, instead of `raise err` (which causes an ERROR) or `continue` (which silently ignores the problem).

Reply:
OK, a FAIL it is.

Reply:
Change made in 5df1372bf74cfc92006fc30c45b42fcddcb0cf92